site/docs/glossary.md

```diff
@@ -0,0 +1,537 @@
+transitively depends on are verified "up to date".
+
+#### Build setting
+
+A Starlark-defined piece of configuration. Transitions can set build settings to
+change a subgraph's configuration. If exposed to the user as a command-line
+flag, also known as a *build flag*.
+
+#### Clean build
+
+A build that doesn't use the results of earlier builds. This is generally slower
+than an *incremental build* but commonly considered to be more correct. Bazel
+guarantees both clean and incremental builds are always correct.
+
+#### Client-server model
+
+The bazel command-line client automatically starts a background server on the
+local machine to execute Bazel commands. The server persists across commands but
+automatically stops after a period of inactivity (or explicitly via bazel
+shutdown). Splitting Bazel into a server and client helps amortize JVM startup
+time and supports faster *incremental builds* because the *action graph* remains
+in memory across commands.
+
+#### Command
+
+Used on the command line to invoke different Bazel functions, like bazel build,
+bazel test, bazel run, and bazel query.
+
+#### Command flags
+
+A set of flags specific to a command. Command flags are specified *after* the
+command (bazel build <command flags>). Flags can be applicable to one or more
+commands. For example, `--configure` is a flag exclusively for the bazel sync
+command, but `--keep_going` is applicable to sync, build, test and more. Flags
+are often used for configuration purposes, so changes in flag values can cause
+Bazel to invalidate in-memory graphs and restart the analysis phase.
+
+#### Configuration
+
+Information outside of rule definitions that impacts how rules generate actions.
+Every build has at least one configuration specifying the target platform,
+action environment variables, and command-line build flags. *Transitions* may
+create additional configurations, e.g. for host tools or cross-compilation.
+
+#### Configuration trimming
+
+The process of only including the pieces of configuration a target actually
+needs. For example, if you build Java binary `//:j` with C++ dependency `//:c`,
+it's wasteful to include `--javacopt` in `//:c`,'s configuration because
+changing `--javacopt` unnecessarily breaks C++ build cacheability.
+
+#### Configured query (cquery)
+
+A query tool that queries over *configured targets* (after the analysis phase
+completes). This means select() and build flags (e.g. `--platforms`) are
+accurately reflected in the results.
+
+#### Configured target
+
+The result of evaluating a *target* with a *configuration*.. The analysis phase
+produces this by combining the build's options with the targets that need to be
+built. For example, if `//:foo` builds for two different architectures in the
+same build, it has two configured targets: `<//:foo, x86>` and `<//:foo, arm>`.
+
+#### Correctness
+
+The property that a build's output faithfully reflects the state of its
```

Comment:
I'd say something like this to make it a bit more clear:

"A build is correct when its output faithfully reflects the state of its transitive inputs."

... and then replace the second sentence with something meaningful about hermeticity and non-determinism?